ee/src/main/java/org/jboss/as/ee/subsystem/ManagedThreadFactoryAdd.java

```diff
@@ -52,13 +50,16 @@ protected void performRuntime(OperationContext context, ModelNode operation, Mod
         final int priority = ManagedThreadFactoryResourceDefinition.PRIORITY_AD.resolveModelAttribute(context, model).asInt();
 
         final ManagedThreadFactoryService service = new ManagedThreadFactoryService(name, jndiName, priority);
-        final ServiceBuilder<ManagedThreadFactoryImpl> serviceBuilder = context.getServiceTarget().addService(ConcurrentServiceNames.getManagedThreadFactoryServiceName(name), service);
+        final ServiceBuilder serviceBuilder = context.getCapabilityServiceTarget()
+                .addService(ManagedThreadFactoryResourceDefinition.CAPABILITY.getCapabilityServiceName(name))
+                .setInstance(service);
+
         String contextService = null;
         if(model.hasDefined(ManagedThreadFactoryResourceDefinition.CONTEXT_SERVICE)) {
             contextService = ManagedThreadFactoryResourceDefinition.CONTEXT_SERVICE_AD.resolveModelAttribute(context, model).asString();
         }
         if (contextService != null) {
-            serviceBuilder.addDependency(ConcurrentServiceNames.getContextServiceServiceName(contextService), ContextServiceImpl.class, service.getContextServiceInjector());
+            serviceBuilder.addDependency(context.getCapabilityServiceName(ContextServiceResourceDefinition.CAPABILITY.getName(), contextService, ContextServiceImpl.class), ContextServiceImpl.class, service.getContextServiceInjector());
```

Comment:
Here, you would want to use (note that serviceBuilder is a CapabilityServiceBuilder<?>):
```java
  serviceBuilder.addCapabilityRequirement(ContextServiceResourceDefinition.CAPABILITY.getName(), ContextServiceImpl.class, service.getContextServiceInjector(), contextService);
```

Reply:
done